exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestAggregateFunctions.java

```diff
@@ -1239,4 +1239,17 @@ public void testInjectVariablesHashAgg() throws Exception {
       client.resetSession(PlannerSettings.STREAMAGG.getOptionName());
     }
   }
+
+  @Test //DRILL-7931
+  public void testRowTypeMissMatch() throws Exception {
+    testBuilder()
+      .sqlQuery("select col1, stddev(col2) as g1, SUM(col2) as g2 FROM " +
+        "(values ('UA', 3), ('USA', 2), ('UA', 3), ('USA', 5), ('USA', 1), " +
+        "('UA', 9)) t(col1, col2) GROUP BY col1 order by col1")
+      .ordered()
+      .baselineColumns("col1", "g1", "g2")
+      .baselineValues("UA", 3.4641016151377544, 15L)
+      .baselineValues("USA", 2.0816659994661326, 8L)
```

Comment:
Thanks for providing a test! The floating point value is likely unstable: it might differ in the last few digits on different systems. I suggest using a function to round the value to, say, four or five (or whatever) decimals to avoid floating point "noise."

Reply:
There is a mechanism built into the test builder for this, you can call approximateEquality(tolerance) to avoid changing the SQL of the test to accomplish what you want.

Reply:
Hi Jason, welcome back to Drill! Thanks much for this suggestion.

Reply:
Thanks Jason, by the way, do you know why approximateEquality is only enabled for unOrdered case, and tolerance is always set to 0 for ordered case?

> There is a mechanism built into the test builder for this, you can call approximateEquality(tolerance) to avoid changing the SQL of the test to accomplish what you want.

Reply:
@Leon-WTF I can't think of a good reason it should be, we may just need to enhance the test builder/runner framework to support this config as well.